src/TypeShape.SourceGenerator/TypeShapeIncrementalGenerator.cs

```diff
@@ -22,20 +22,30 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 "TypeShape.GenerateShapeAttribute<T>",
                 (node, _) => node is ClassDeclarationSyntax)
             .Combine(knownSymbols)
-            .Select((tuple, cancellationToken) => 
+            .Select((tuple, cancellationToken) =>
                 Parser.ParseFromGenerateShapeOfTAttributes(
                     context: tuple.Left,
-                    knownSymbols: tuple.Right, 
+                    knownSymbols: tuple.Right,
                     cancellationToken));
 
+
+        // I kinda expect this to break the whole incremental pipeline
```

Comment:
My expectation here would be that this should be fused with the incremental values defined in line 39. The `ForTypeWithAttributeDeclaration` helper could be augmented with an optional `GlobalOptions` key so that the search is restricted to one pass (and should serve to collapse duplicates in cases where a type is annotated with both `GenerateShape` and the user-defined attribute). 

Such a change would necessitate lifting the extension method from the syntax provider to the `IncrementalGeneratorInitializationContext` itself so that the helper can access the `AnalyzerConfigOptionsProvider`.